carball/analysis/stats/possession/per_possession.py

```diff
@@ -0,0 +1,251 @@
+                    player_possessions.append(current_player_possession)
+                    current_player_possession = PlayerPossession([hit], player_id)
+
+        return player_possessions, team_possessions
+
+    @staticmethod
+    def add_possession_durations(possessions: List['BasePossession'], data_frame: pd.DataFrame):
+        player_possessions_count = len(possessions)
+        for i, possession in enumerate(possessions):
+            first_hit = possession.hits[0]
+            possession_start_frame = first_hit.frame_number
+
+            if i != player_possessions_count - 1:
+                # Next possession exists
+                next_possession = possessions[i + 1]
+                if next_possession.hits[0].goal_number != first_hit.goal_number:
+                    # Next possession is in next goal frame
+                    # possession end frame is end of goal frame
+                    possession_end_frame = \
+                        data_frame.index[(data_frame.loc[:, ('game', 'goal_number')] == first_hit.goal_number)][-1]
+                else:
+                    # possession end frame is first hit of next possession
+                    possession_end_frame = next_possession.hits[0].frame_number
+            else:
+                # Last possession of game
+                # possession end frame is last frame of game
+                possession_end_frame = data_frame.index.max()
+
+            possession_times = data_frame.loc[[possession_start_frame, possession_end_frame], ('game', 'time')].values
+            possession_duration = possession_times[1] - possession_times[0]
+            possession.duration = possession_duration
+
+    @staticmethod
+    def add_possession_counts(possessions: List['BasePossession']):
+        fields = [
+            'pass_', 'passed',
+            'dribble', 'dribble_continuation',
+            'shot', 'goal',
+            'assist', 'assisted',
+            'save',
+            'aerial',
+        ]
+        for possession in possessions:
+            for hit in possession.hits:
+                for field in fields:
+                    if getattr(hit, field):
+                        possession.counts[field] += 1
+
+    @staticmethod
+    def add_possession_distances(possessions: List['BasePossession']):
+        for possession in possessions:
+            distance = 0
+            for hit in possession.hits:
+                distance += hit.distance
+            possession.distance = distance
+
+    @classmethod
+    def get_per_possession_stats(cls, possessions: List['BasePossession']) -> PerPossessionStats:
+        count = len(possessions)
+        hit_counts = [len(possession.hits) for possession in possessions]
+        average_hits = sum(hit_counts) / len(hit_counts)
+        durations = [possession.duration for possession in possessions]
+        average_duration = sum(durations) / len(durations)
+
+        average_counts = {}
+        fields = [
```

Comment:
this list exist in multiple places.
Can we find a way to only exist once and then reference it?